common/flatpak-run.c

```diff
@@ -4553,21 +4511,16 @@ setup_seccomp (GPtrArray  *argv_array,
         seccomp_rule_add_exact (seccomp, SCMP_ACT_ERRNO (EAFNOSUPPORT), SCMP_SYS (socket), 1, SCMP_A0 (SCMP_CMP_EQ, family));
     }
 
-  fd = g_file_open_tmp ("flatpak-seccomp-XXXXXX", &path, error);
-  if (fd == -1)
+  if (!glnx_open_anonymous_tmpfile (O_RDWR | O_CLOEXEC, &seccomp_tmpf, error))
     return FALSE;
 
-  unlink (path);
-
-  if (seccomp_export_bpf (seccomp, fd) != 0)
+  if (seccomp_export_bpf (seccomp, seccomp_tmpf.fd) != 0)
     return flatpak_fail (error, "Failed to export bpf");
 
-  lseek (fd, 0, SEEK_SET);
+  lseek (seccomp_tmpf.fd, 0, SEEK_SET);
 
   add_args_data_fd (argv_array, fd_array,
-                    "--seccomp", fd, NULL);
-
-  fd = -1; /* Don't close on success */
+                    "--seccomp", glnx_steal_fd (&seccomp_tmpf.fd), NULL);
```

Comment:
We're leaking the fd here.  Really what we need to do is to have a "struct RunContext" or so that has-a `GSubprocessLauncher`, and then we can use `g_subprocess_launcher_take_fd()` which has all the semantics we want.

But eh...this isn't a new issue, since before we had
`fd = -1; /* Don't close on success */`
which did the same thing.